src/modules/tensor/rppt_tensor_effects_augmentations.cpp

```diff
@@ -1347,6 +1347,66 @@ RppStatus rppt_rain_host(RppPtr_t srcPtr,
     return RPP_SUCCESS;
 }
 
+/******************** posterize ********************/
+
+RppStatus rppt_posterize_host(RppPtr_t srcPtr,
+                              RpptDescPtr srcDescPtr,
+                              RppPtr_t dstPtr,
+                              RpptDescPtr dstDescPtr,
+                              Rpp32u *posterizeLevelBits,
+                              RpptROIPtr roiTensorPtrSrc,
+                              RpptRoiType roiType,
+                              rppHandle_t rppHandle)
+{
+    if ((srcDescPtr->layout != RpptLayout::NCHW) && (srcDescPtr->layout != RpptLayout::NHWC)) return RPP_ERROR_INVALID_SRC_LAYOUT;
+    if ((dstDescPtr->layout != RpptLayout::NCHW) && (dstDescPtr->layout != RpptLayout::NHWC)) return RPP_ERROR_INVALID_DST_LAYOUT;
+    for(int i = 0; i < srcDescPtr->n; i++)
+        if((posterizeLevelBits[i] < 0) && (posterizeLevelBits[i] > 8))
```

Comment:
Arguments like posterizeLevelBits can say pinned/HIP memory in the header docs only if both are acceptable.
If there are array accesses like this on the host side, giving HIP memory and accessing outside hip kernels will give seg fault. Pls check